### DISCUSSION TITLE
Assistance Required for Integrating Phantom Wallet with Web3 App in an iFrame

### DISCUSSION THREAD

**Discussion by brauliolomeli** [phantom/docs#151] [Help & Support] [answered]
### Summary

I've hit a snag with running Phantom in an iframe for my web3 app. I get why you guys avoid iframes for security, but I really need this to work. Other wallets are fine, but Phantom's giving me trouble.

Is there any workaround, considering I control both the host and the iframe?  Would really appreciate your help!

### Example

https://phantom-2.dev.apps.dappslap.ai/embedded.html

### Steps to Reproduce

1. Embed the web app in an iframe
2. try to connect Phantom.

### Phantom Version

Phantom Desktop

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hi @brauliolomeli  We do support iFrames, in general. You can see an example by going to Jupiter and viewing their limit order tab. 

Also I was able to connect to your example without issue. Can you further describe the issue you're having?

**Reply by brauliolomeli** [reply]
I saw in multiple places [in the docs](https://docs.phantom.app/solana/detecting-the-provider) that _Phantom will not inject the provider into iframes_

I've tried the Google Chrome and Brave extensions getting the same result, when it's on an [iframe](https://phantom-2.dev.apps.dappslap.ai/embedded.html) it gets stuck in a `connecting` state but it works fine opening the app [directly](https://phantom-2.dev.apps.dappslap.ai/) :

<img width="300" alt="image" src="https://github.com/phantom/docs/assets/8377819/943502b6-905a-405e-a62a-599abff8156e">

The example was created using [@solana/wallet-adapter-react-ui](https://www.npmjs.com/package/@solana/wallet-adapter-react-ui)


Here's another [example](https://codepen.io/Braulio-Lomeli/pen/WNmwvVN) embedding the [Solana wallet adapter demo page](https://solana-labs.github.io/wallet-adapter/example/), when I select Phantom it also gets stuck in a 'connecting' state:

<img width="300" alt="image" src="https://github.com/phantom/docs/assets/8377819/08df6eb4-9f6c-40ff-90b0-aeef179d4ec0">

I took a look at the Jupiter limit order tab but doesn't seem to have any iframes other than the chart view, which doesn't interact with a wallet.

**Reply by adamdelphantom** [reply, marked as answer]
@brauliolomeli Thank you for the clarification. I was able to connect on Phantom mobile but not on Phantom extension. I will review this internally with our engineering team but as it is mentioned in the docs it will likely not be supported.

**Reply by adrianotadao** [reply]
hey @adamdelphantom any news on this? I am facing the exactly same problem

**Reply by adamdelphantom** [reply]
@adrianotadao No updates. I would suggest developing your dApp with the assumption that it will not be possible to access the provider in an iFrame.

**Reply by 0x4Graham** [reply]
@adamdelphantom Please consider supporting iFrame's. Going through the forum, lots of other developers are running into the same blocker.

**Reply by adamdelphantom** [reply]
@0x4Graham I can confirm that there are no plans to make the provider available within an iframe. Apologies for the difficulty this provides but it is out of an abundance of caution for the safety of Phantom users.
